lib/ret_web/controllers/page_controller.ex

```diff
@@ -205,23 +200,43 @@ defmodule RetWeb.PageController do
 
   def render_hub_content(conn, hub, _slug) do
     hub = hub |> Repo.preload(scene: [:screenshot_owned_file], scene_listing: [:screenshot_owned_file])
+    {app_config_script, app_config_csp} = generate_app_config()
 
     hub_meta_tags =
       Phoenix.View.render_to_string(RetWeb.PageView, "hub-meta.html",
         hub: hub,
         scene: hub.scene,
-        ret_meta: Ret.Meta.get_meta(include_repo: false)
+        ret_meta: Ret.Meta.get_meta(include_repo: false),
+        app_config_script: {:safe, app_config_script}
       )
 
     chunks =
       chunks_for_page("hub.html", :hubs)
       |> List.insert_at(1, hub_meta_tags)
 
     conn
+    |> append_csp("script-src", app_config_csp)
     |> put_resp_header("content-type", "text/html; charset=utf-8")
     |> send_resp(200, chunks)
   end
 
+  defp generate_app_config() do
+    app_config =
```

Comment:
I'd move this to `Ret.AppConfig`

Reply:
Huh, I thought exactly the opposite, since the method deals with caching (skip_cache particularly), CSP and HTML. Why do you prefer it to be in AppConfig?

Reply:
ah, I guess i just have a general aversion to any 'controller' method that theoretically could be stuffed somewhere else. i think you're right though that the caching bits don't belong there. it was the script/csp generation that caught my eye. i don't mind if you keep it here.